Modules/HRV_DFA/DFA.cpp

```diff
@@ -0,0 +1,145 @@
+#include "DFA.h"
+
+using namespace std;
+
+void polyfit::calculateCoefficient(vector<int>& x)
+    {
+        double N = x.size();
+        double numerator
+            = (N * sum_xy - sum_x * sum_y);
+        double denominator
+            = (N * sum_x_square - sum_x * sum_x);
+        coeff = numerator / denominator;
+    }
+
+void polyfit::calculateConstantTerm(vector<int>& x)
+    {
+        double N = x.size();
+        double numerator
+            = (sum_y * sum_x_square - sum_x * sum_xy);
+        double denominator
+            = (N * sum_x_square - sum_x * sum_x);
+        constTerm = numerator / denominator;
```

Comment:
Możliwe, że nie jest to zbyt realny scenariusz, ale co jeżeli `denominator == 0`?

Reply:
denominator nigdy nie będzie równy 0, bo pracujemy na double'u z liczbami różnymi od 0. Ten algorytm i tak finalnie powinien być zmieniony na obliczanie macierzowe

Reply:
zmienione i jw - nie ma opcji dzielenia przez 0